g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/StreamProductAction.java

```diff
@@ -0,0 +1,174 @@
+		writeTranslationsToChanel(readChannels, wbc);
+		 if(isPartContent){
+			 wbc.write(ByteBuffer.wrap(addNullMessage(readChannels, componentList, localeList)));
+		 }
+		 wbc.write(sr.getEndBytes());
+	}
+
+	private boolean writeResponseHeader(boolean notVersionFallback, int expSize, int readChannelSize, WritableByteChannel wbc, StreamProductResp sr) throws IOException {
+		boolean isPartContent = false;
+		if(notVersionFallback) {
+			isPartContent = (readChannelSize != expSize);
+			if(isPartContent) {
+				wbc.write(sr.getRespStartBytes(APIResponseStatus.MULTTRANSLATION_PART_CONTENT.getCode()));
+			}else {
+				wbc.write(sr.getRespStartBytes(APIResponseStatus.OK.getCode()));
+			}
+		}else {
+			wbc.write(sr.getRespStartBytes(APIResponseStatus.VERSION_FALLBACK_TRANSLATION.getCode()));
+		}
+		return isPartContent;
+	}
+	
+	
+	private List<String> getcomponentList(String productName, String versionStr, String components) throws L3APIException{
+		List<String> componentList = null;
+		if (StringUtils.isEmpty(components)) {
+			componentList = productService.getComponentNameList(productName, versionStr);
+		} else {
+			componentList = new ArrayList<String>();
+			for (String component : components.split(",")) {
+				componentList.add(component.trim());
+			}
+		}
+		
+		return componentList;
+	}
+	
+	private List<String> getlocaleList(String productName, String versionStr, String locales, String pseudo) throws L3APIException{
+		List<String> localeList = new ArrayList<String>();
+		if (Boolean.valueOf(pseudo)) {
+			localeList.add(ConstantsKeys.LATEST);
+		} else if (!StringUtils.isEmpty(locales)) {
+			for (String locale : locales.split(",")) {
+				localeList.add(super.getMappingLocale(productName, versionStr, locale.trim()));
+			}
+		} else {
+			localeList = productService.getSupportedLocaleList(productName, versionStr);
+		}
+		
+		return localeList;
+	}
+	
+	private byte[] addNullMessage(List<ResultMessageChannel>  readChannels, List<String> components, List<String> locales) {
+		StringBuilder sb = new StringBuilder();
+		for (String component : components) {
+			for (String locale : locales) {
+				ResultMessageChannel rmc = new ResultMessageChannel(component, locale, null);
+				if(!readChannels.contains(rmc)) {
+					sb.append(",");
+					sb.append(rmc.generateNullMessage());
+				}
+				
+			}
+		}
+		return sb.toString().getBytes();
+	}
+  
+
+
+	private void transferTo(ReadableByteChannel readChannel, WritableByteChannel writeChannel, ByteBuffer buf) throws Exception {
+
+		if (readChannel instanceof FileChannel) {
+			try (FileChannel fileChannel = (FileChannel) readChannel) {
+				fileChannel.transferTo(0, fileChannel.size(), writeChannel);
+			}
+		} else {
+			
+			try {
+				while ((readChannel.read(buf)) != -1) {
+					buf.flip();
+					do {
+						writeChannel.write(buf);
+					}while (buf.hasRemaining());
+					
+					buf.clear();
+				}
+			} catch (Exception e) {
+				throw e;
```

Comment:
log error?

Reply:
Because have exception handler, Colin suggests that remove the error log